packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -216,6 +239,26 @@ export class WaferMap extends FoundationElement {
     private canvasHeightChanged(): void {
         this.queueRender();
     }
+
+    private hoverDieChanged(): void {
+        this.$emit('die-hover', { currentDie: this.hoverDie });
+        this.hoverOpacity = this.hoverDie === undefined
+            ? HoverDieOpacity.hide
+            : HoverDieOpacity.show;
+        this.moveHoverRect();
+    }
+
+    private moveHoverRect(): void {
+        if (this.hoverDie !== undefined) {
+            const scaledX = this.dataManager!.horizontalScale(this.hoverDie.x);
+            const scaledY = this.dataManager!.verticalScale(this.hoverDie.y);
+            const transformedPoint = this.transform.apply([
+                scaledX + this.dataManager!.margin.left,
+                scaledY + this.dataManager!.margin.top
+            ]);
+            this.hoverTransform = `translate(${transformedPoint[0]}, ${transformedPoint[1]})`;
+        }
+    }
```

Comment:
I'm surprised to see this hover specific logic in the top-level index.ts and not following the existing conventions.

The data flow I expect based on existing patterns is propertyUpdate -> queueRender -> render.

I can appreciate there is an evaluation order where:
mouse move -> calculate the hovered die -> set hoverDie -> queueRender -> render (scaleCanvas, renderDies, renderText, renderHover)

So I would have expected an organization like:
```ts
// index.ts:

private hoverDieChanged(): void {
    this.$emit('die-hover', { currentDie: this.hoverDie });
    this.queueRender();
}


// rendering.ts

public drawWafer (): {
     // draw other parts
    this.renderHover();
}

renderHover (): {
    waferMap.hoverWidth = 
    waferMap.hoverHeight = 
    waferMap.hoverOpacity = 
    waferMap.hoverTransform = 
}

```

ie the hoverDie represents the state of the hover behavior and the other properties represent rendered results based on that state and the other existing state (hoverWidth, hoverHeight, hoverOpacity, hoverTransform).